crates/ruff_diagnostics/src/diagnostic.rs

```diff
@@ -40,8 +42,11 @@ impl Diagnostic {
         }
     }
 
-    pub fn amend(&mut self, fix: Fix) -> &mut Self {
-        self.fix = Some(fix);
+    pub fn amend(&mut self, fix: Result<Fix>) -> &mut Self {
+        match fix {
+            Ok(fix) => self.fix = Some(fix),
+            Err(err) => error!("Failed to create fix: {}", err),
+        }
         self
     }
```

Comment:
Handle logging errors from failed auto-fix here.

Reply:
I'd like to get @MichaReiser's opinion, it strikes me as potentially unusual to have an API accept `Result` (since, in _most_ cases, we know the value is `Ok`), though the ergonomic benefits are clear.

Without looking at other APIs for inspiration... perhaps something like this would be more idiomatic?

```rust
pub fn try_amend(&mut self, fn: FnOnce() -> Result<Fix>) {
  ...
}
```

Don't act on that proposal yet though, since @MichaReiser may have a different suggestion.

Reply:
I am undecided on the API. Mainly because I don't yet understand how we want to support this in the long-term where we shouldn't use `error` if a fix fails but instead add a diagnostic to the output that then gets emitted the same as any other diagnostic. 

But that's still far in the future and it should not prevent us from improving the API in the short term. I like @charliermarsh's suggestion of adding a second `try_amend` that accepts a callback. I find this more idiomatic than passing `Result` but mainly like it because it doesn't require changing all `amend` calls where the parameter always is `Ok` 


Unrelated: What's the reason for the name `amend`. How about renaming the function to `with_fix` or `try_with_fix` in another PR?

Reply:
Yeah good call. I'll handle the rename.